src/ast/ast_validations.c

```diff
@@ -411,6 +562,13 @@ static AST_Validation _AST_Validate(const AST *ast, char **reason) {
     return AST_VALID;
 }
 
+AST_Validation AST_WhitelistQuery(const cypher_astnode_t *root, char **reason) {
+    rax *whitelist = _BuildWhitelist();
```

Comment:
we're constructing `whitelist` for each query, as `whitelist` not going to change from one query to the other (same white-list) construct it once, and reuse it. (declare it in global scope as static)